packages/grafana-runtime/src/services/LocationService.ts

```diff
@@ -134,15 +135,30 @@ class HistoryWrapper implements LocationService {
  * @alpha
  * Parses a location search string to an object
  * */
-export function locationSearchToObject(search: string): UrlQueryMap {
-  if (search.length > 0) {
-    return urlUtil.parseKeyValue(search.substring(1));
+export function locationSearchToObject(search: string | number): UrlQueryMap {
+  let queryString = typeof search === 'number' ? String(search) : search;
+
+  if (queryString.length > 0) {
+    if (queryString.startsWith('?')) {
+      return urlUtil.parseKeyValue(queryString.substring(1));
+    }
+    return urlUtil.parseKeyValue(queryString);
   }
 
   return {};
 }
 
-export const locationService: LocationService = new HistoryWrapper();
+export let locationService: LocationService = new HistoryWrapper();
+
+/** @internal
+ * Used for tests only
+ */
+export const setLocationService = (location: LocationService) => {
```

Comment:
@torkelo added this internal setter for test purposes. I want a clean history in every test and this seems to be a reasonable way to do it.

Reply:
:+1: